### PR TITLE
Add Spack CPU build with `exago+hiop+raja~ipopt ^hiop+raja~sparse`

### LINKED ISSUE #40: `exago+hiop+raja~ipopt ^hiop+raja~sparse` fails to build
https://github.com/xsdk-project/xsdk-issues/issues/226 - thanks to @balos1 for catching this.

Copying from the xSDK issue:

> @abhyshr and @wperkins this error is occuring with a build of exago+mpi+python+ipopt~rocm~cuda. Key here is ~ipopt means https://github.com/pnnl/ExaGO/blob/828db06af5e0345b641b6d03ef7e2456d13469ea/src/opflow/interface/opflowregi.cpp#L79 and https://github.com/pnnl/ExaGO/blob/828db06af5e0345b641b6d03ef7e2456d13469ea/src/opflow/model/power_bal_hiop/pbpolrajahiopsparse.cpp#L3 end up being guarded by different compiler macros.

### PR DESCRIPTION
**Pull request by cameronrutherford**
With the additional GitHub action we should be able to reproduce issue identified in xSDK testing, so we can add a fix that is verifiable in local pipelines.

Closes:
- #40 

xSDK issues:
- https://github.com/xsdk-project/xsdk-issues/issues/231
- https://github.com/xsdk-project/xsdk-issues/issues/226

### CODE REVIEW

File: .github/workflows/spack_cpu_build.yaml [hunk 1]
```diff
@@ -21,6 +21,8 @@ jobs:
           # See #16 - +python~mpi causes issues
         # - exago@develop~mpi~ipopt+hiop+python~raja
           - exago@develop+mpi~ipopt+hiop+python~raja ^openmpi
+        # Builds with hiop sparse and without Ipopt are useful for edge cases
+          - exago@develop~mpi~ipopt+hiop~python+raja ^hiop+raja+sparse
```
Comment (cameronrutherford) [inline .github/workflows/spack_cpu_build.yaml:26]:
```suggestion
        # Builds with hiop and without Ipopt are useful for edge cases
          - exago@develop~mpi~ipopt+hiop~python+raja ^hiop+raja~sparse

```

**Comment by cameronrutherford** [comment]
Added `hiop+sparse` initially, which both doesn't work in GitHub actions, and doesn't repro the issue... Re-trying with the correct build

**Comment by cameronrutherford** [comment]
I re-produced #18... Going to add build with MPI and try to repro original issue #40

**Comment by cameronrutherford** [comment]
Successfully reproduced the issue.  Just need to write the fix

**Comment by cameronrutherford** [comment]
@wperkins and @abhyshr please review and make sure I am `#ifdef` guarding the right things here. If GitHub actions (and other pipelines) pass then we are good to merge and close the xSDK bug.

**Review by wperkins — APPROVED** [review]
This looks fine to me.

**Pull request merged**